lib/ansible/modules/cloud/misc/virt.py

```diff
@@ -450,6 +461,11 @@ def core(module):
             module.fail_json(msg = "state change requires a guest specified")
 
         res['changed'] = False
+        autostart_res = v.autostart(guest, autostart)
+        if autostart_res is not False:
+            res['changed'] = True
```

Comment:
`res['changed']` and `res['msg']` values will be lost/overwritten when the current state isn't the wanted state.